.github/workflows/pr.yml

```diff
@@ -0,0 +1,48 @@
+name: PR
+on:
+  pull_request:
+    branches:
+      - 'main'
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21.0
+      - name: go fmt
+        run: if [ "$(gofmt -s -l . | wc -l)" -gt 0 ]; then exit 1; fi
+  test:
+    runs-on: ubuntu-latest
+    needs: lint
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
+      - run: go test -v ./...
```

Comment:
Does this run the tests only in scope that the go modules is? (i.e. does it _not_ run the examples tests?)

Reply:
Yes exactly, it only runs the tests in the scope that the go modules is in. If you look at the pipeline run only `TestCompareField`, `TestParseMapToStruct` and `TestService_Validate` Run.